app/src/sharedTest/java/org/oppia/android/app/faq/FaqListActivityTest.kt

```diff
@@ -0,0 +1,145 @@
+import org.oppia.android.domain.classify.rules.fractioninput.FractionInputModule
+import org.oppia.android.domain.classify.rules.imageClickInput.ImageClickInputModule
+import org.oppia.android.domain.classify.rules.itemselectioninput.ItemSelectionInputModule
+import org.oppia.android.domain.classify.rules.multiplechoiceinput.MultipleChoiceInputModule
+import org.oppia.android.domain.classify.rules.numberwithunits.NumberWithUnitsRuleModule
+import org.oppia.android.domain.classify.rules.numericinput.NumericInputRuleModule
+import org.oppia.android.domain.classify.rules.ratioinput.RatioInputModule
+import org.oppia.android.domain.classify.rules.textinput.TextInputRuleModule
+import org.oppia.android.domain.onboarding.ExpirationMetaDataRetrieverModule
+import org.oppia.android.domain.oppialogger.LogStorageModule
+import org.oppia.android.domain.oppialogger.loguploader.LogUploadWorkerModule
+import org.oppia.android.domain.oppialogger.loguploader.WorkManagerConfigurationModule
+import org.oppia.android.domain.question.QuestionModule
+import org.oppia.android.domain.topic.PrimeTopicAssetsControllerModule
+import org.oppia.android.testing.RobolectricModule
+import org.oppia.android.testing.TestAccessibilityModule
+import org.oppia.android.testing.TestDispatcherModule
+import org.oppia.android.testing.TestLogReportingModule
+import org.oppia.android.testing.time.FakeOppiaClockModule
+import org.oppia.android.util.caching.testing.CachingTestModule
+import org.oppia.android.util.gcsresource.GcsResourceModule
+import org.oppia.android.util.logging.LoggerModule
+import org.oppia.android.util.logging.firebase.FirebaseLogUploaderModule
+import org.oppia.android.util.parser.GlideImageLoaderModule
+import org.oppia.android.util.parser.HtmlParserEntityTypeModule
+import org.oppia.android.util.parser.ImageParsingModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Tests for [FAQListActivity]. */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = FaqListActivityTest.TestApplication::class, qualifiers = "port-xxhdpi")
+class FaqListActivityTest {
+
+  @get:Rule
+  val activityTestRule: ActivityTestRule<FAQListActivity> = ActivityTestRule(
+    FAQListActivity::class.java, /* initialTouchMode= */ true, /* launchActivity= */ false
+  )
+
+  @Inject
+  lateinit var context: Context
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testFaqListActivity_hasCorrectActivityLabel() {
+    activityTestRule.launchActivity(createFaqListActivityIntent())
+    val title = activityTestRule.activity.title
+
+    // Verify that the activity label is correct as a proxy to verify TalkBack will announce the
+    // correct string when it's read out.
+    assertThat(title).isEqualTo(context.getString(R.string.faq_activity_label))
```

Comment:
@rt4914 random thought: given we keep repeating this context in a bunch of places, could we maybe introduce an extension method to contextualize this? E.g. introducing a flow that would let us do:

```kotlin
assertThat(activityTestRule.activity).hasTalkBackLabel(R.string.faq_activity_label)
```

I was thinking at first that we could maybe do this with an extension function, but I actually think we will need to implement a custom subject (https://truth.dev/extension.html) since unfortunately AndroidX doesn't include one for Activity (https://developer.android.com/training/testing/fundamentals#assertions). This might lead to other nice wins when testing activities, and we could establish other extensions as we need them. Might be worth filing an issue to track. @anandwana001 as well since this falls under testing.

Note that I do not suggest doing any of this work in this PR--let's keep the current pattern & consider doing the above.

Reply:
Sounds good. @anandwana001 Can you file an issue for this?

Reply:
@BenHenning is Custome Subject is available in Kotlin?
Reference - https://github.com/google/truth/issues/536

I just had a quick look over what a custom subject is and tried this gist, could you check once if it is the correct way?
https://gist.github.com/anandwana001/9bccce1f9a0413af5f86ec12dc331d24

I will look into more on this today and see how to have a clean implementation.

Reply:
Custom subjects should work fine in Kotlin. Did you observe something that suggests they won’t?

Reply:
(From what I can tell the issue you linked to is for using extensions which is actually why I was thinking custom subjects originally).